src/ts/ConvivaAnalytics.ts

```diff
@@ -130,51 +135,78 @@ export class ConvivaAnalytics {
     }
   }
 
-  private startSession = (event?: any) => {
-    let source = this.player.getConfig().source;
+  private initializeSession() {
+    this.contentMetadata = new Conviva.ContentMetadata();
+    this.contentMetadata.streamType = Conviva.ContentMetadata.StreamType.UNKNOWN;
+    this.contentMetadata.applicationName = this.config.applicationName || 'Unknown (no config.applicationName set)';
 
-    let assetId = source.contentId ? `[${source.contentId}]` : undefined;
-    let assetTitle = source.title;
-    let assetName;
+    let source = this.player.getConfig().source;
+    if (source) {
+      this.contentMetadata.assetName = this.getAssetName(source);
+      this.contentMetadata.streamUrl = this.getUrlFromSource(source);
 
-    if (assetId && assetTitle) {
-      assetName = `${assetId} ${assetTitle}`;
-    } else if (assetId && !assetTitle) {
-      assetName = assetId;
-    } else if (assetTitle && !assetId) {
-      assetName = assetTitle;
     } else {
-      assetName = 'Untitled (no source.title/source.contentId set)';
+      this.contentMetadata.assetName = 'unknown';
+      this.contentMetadata.streamUrl = 'unknown';
```

Comment:
Wouldn't it be cleaner to not report unknown data at all instead of reporting it, thus giving the system the impression that we know it, but then writing in the content that we don't know?

Reply:
I was sticking to the documentation, which states that these fields are required. 'https://community.conviva.com/site/global/platforms/other_platforms/sdks/javascript_sdk/taskref/index.gsp#metadata_ref'

Reply:
Ok then the question is what required means... required from the beginning or required to be set sometime during the session? If the latter, then you can still set it `unknown` in `ON_SOURCE_LOADED` where it should actually be known.